src/main/java/com/team4/testingsystem/converters/TestConverter.java

```diff
@@ -0,0 +1,58 @@
+package com.team4.testingsystem.converters;
+
+import com.team4.testingsystem.dto.QuestionDTO;
+import com.team4.testingsystem.dto.TestDTO;
+import com.team4.testingsystem.entities.ContentFile;
+import com.team4.testingsystem.entities.Question;
+import com.team4.testingsystem.entities.Test;
+import com.team4.testingsystem.enums.Modules;
+import com.team4.testingsystem.services.ContentFilesService;
+import com.team4.testingsystem.services.QuestionService;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Component;
+
+import java.util.List;
+import java.util.stream.Collectors;
+
+@Component
+public class TestConverter {
+
+    private final QuestionService questionService;
+    private final ContentFilesService contentFilesService;
+
+    @Autowired
+    public TestConverter(QuestionService questionService,
+                         ContentFilesService contentFilesService) {
+        this.questionService = questionService;
+        this.contentFilesService = contentFilesService;
+    }
+
+    public TestDTO convertToDTO(Test test) {
```

Comment:
Let's split this method into smaller ones, like
* `generateQuestions`, where all questions are set
* `appendContentFiles`
So, this method will create TestDTO and call other methods to complete the DTO